Cli/Command/CheckViolationsCommand.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+/*
+ * This file is part of the SensioLabsInsight package.
+ *
+ * (c) Lhassan Baazzi <baazzilhassan@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace SensioLabs\Insight\Cli\Command;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Gitonomy\Git\Repository as GitRepository;
+use Gitonomy\Git\RevisionList as GitRevisionList;
+
+class CheckViolationsCommand extends Command implements NeedConfigurationInterface
+{
+    protected function configure()
+    {
+        $this
+            ->setName('check-violations')
+            ->addArgument('project-uuid', InputArgument::REQUIRED, 'SensioLabs insight project UUID')
+            ->addOption('repository'    , null, InputOption::VALUE_REQUIRED, 'The GIT repository path, default to current working directory', getcwd())
+            ->addOption('commits'       , null, InputOption::VALUE_IS_ARRAY | InputOption::VALUE_REQUIRED, 'List of commits')
+            ->addOption('no-global'     , null, InputOption::VALUE_NONE, 'Show global violations', null)
+            ->addOption('with-assets'   , null, InputOption::VALUE_NONE, 'Ignore assets violations(css, js, coffee, less, sass)', null)
```

Comment:
the name and the description don't match. It would be very weird to use ``--with-assets`` to **ignore** the asset violations